pandas/util/testing.py

```diff
@@ -1185,8 +1201,11 @@ def assert_series_equal(
                 right._internal_get_values(),
                 check_dtype=check_dtype,
             )
-    elif is_interval_dtype(left) or is_interval_dtype(right):
-        assert_interval_array_equal(left.array, right.array)
+    elif is_interval_dtype(left) or is_interval_dtype(left):
+        # must cast to interval dtype to keep mypy happy
+        left_array = cast(IntervalArray, left.array)
```

Comment:
side note @WillAyd if ``.array`` property is typed (will be tricky as it can return a Union of lots of things), this cast might then be uneeded?